config/routes.config.js

```diff
@@ -0,0 +1,35 @@
+//list our routes in the flow order in the app
+//for address edit and marital status edit, I am specifically pointing to data that does not exist, to force skipping the route when navigating back
+const routes = [
+  { name: 'start', path: '/start' },
+  { name: 'login code', path: '/login/code' },
+  { name: 'login sin', path: '/login/sin' },
+  { name: 'login date of birth', path: '/login/dateOfBirth' },
+  { name: 'name', path: '/personal/name' },
+  { name: 'residence', path: '/personal/residence' },
+  { name: 'address', path: '/personal/address' },
+  { name: 'address edit', path: '/personal/address/edit', editInfo: 'skip' },
+  { name: 'income', path: '/financial/income' },
+  { name: 'rrsp', path: '/deductions/rrsp' },
+  { name: 'rrsp amount', path: '/deductions/rrsp/amount', editInfo: 'deductions.rrspClaim' },
+  { name: 'marital status', path: '/personal/maritalStatus' },
+  { name: 'marital status edit', path: '/personal/maritalStatus/edit', editInfo: 'skip' },
+  { name: 'medical', path: '/deductions/medical' },
+  { name: 'medical amount', path: '/deductions/medical/amount', editInfo: 'deductions.medicalExpenseClaim' },
+  { name: 'political', path: '/deductions/political' },
+  { name: 'political amount', path: '/deductions/political/amount', editInfo:'deductions.politicalContributionClaim' },
+  { name: 'donations', path: '/deductions/donations' },
+  { name: 'donations amount', path: '/deductions/donations/amount', editInfo:  'deductions.charitableDonationAmount' },
+  { name: 'trillium rent', path: '/trillium/rent/amount', editInfo: 'deductions.trilliumRentAmount' },
+  { name: 'trillium property tax', path: '/deductions/trillium/propertyTax/amount', editInfo: 'deductions.trilliumPropertyTaxAmount' },
+  { name: 'trillium student residence', path: '/trillium/studentResidence'},
+  { name: 'trillium energy amount', path: '/trillium/energy/amount', editInfo: 'deductions.trilliumEnergyAmount' },
+  { name: 'trillium long term care amount', path: '/trillium/longTermCare/amount', editInfo: 'deductions.trilliumLongTermCareAmount' },
+  { name: 'climate action incentive', path: '/deductions/climate-action-incentive' },
+  { name: 'review', path: '/review' },
+  { name: 'confirmation', path: '/confirmation' },
+]
```

Comment:
I have a potential refactor that breaks compatibility with Tim's code, but hear me out. 

Our "route" dictionaries have two keys: `name` and `path`, both of which are unique. As in, for any given route, the `name` is unique and so is the `path`. 

So finding a route by `name` is the same as finding a route by `path`. 

Therefore, using a name creates a new id-type value that that we didn't have before and is _only_ used as a lookup value. 

SO, (galaxy brain), since we always call `getPreviousRoute` inside of middleware, which has access to the request  object (`req`), which always knows about the url path `req.path`, we shouldn't need to pass another value ever to `renderWithData`. We can just let it figure out the link itself using the request object.

We could potentially just call `getPreviousData` with `req.path`, and then refactor `getRouteWithIndexByName` to find by path rather than name. 

That way

- we don't need the second param in `renderWithData`
- we don't need the `name` key in our routes

Does that all hang together or have I overlooked something?

Reply:
No that checks out and I was thinking about that too. I'll work on it in the AM